docs/vm-configuration.md

```diff
@@ -264,3 +264,95 @@ the generation of multiple libvirt XML documents, the domain XML and associated
 documents for secrets, network filters and node devices. So there is not a strict
 1-1 mapping between the k8s API resource representation and the libvirt low
 level representation.
+
+Proposed VM Definition
+----------------------
+
+The following definition adheres to main points described above. It provides
+
+1. high level description usable by end users or management applications and
+2. low level details required to construct and launch a VM.
+
+The features section is used to generate the domain section, the other way
+around is out of scope for this proposal. Each feature must represent one or
+more fields in the domain section.
+
+The design and contents of each feature section should be logically split and
+discussed in separate proposal. This is just an example without the sections
+discussed, converted from cluster/vm.yaml. Splitting the features enables us to
+create modular VM Spec generation flows that are easily tunable by experts in
+their domains of expertise.
+
+```yaml
+apiVersion: kubevirt.io/v1alpha2
+kind: VM
+metadata:
+  name: testvm
+spec:
+  features:
```

Comment:
Maybe we can carefully phrase it as, _use VM for full controll, or VMConfig for ease and persistence_. Something along those lines.

@rmohr The "classic" admission controllers modify the spec during creation (and thus store a modified spec), to me this is suboptimal for this `feature` feature (ha!).

If a `feature` is specified, then we still need to be able to see in the retrospective, which feature it was. In this two-spec case, it's easy and straight forward.
Technically we could probably push this into the VM spec, but I've got the impression that it will get messy.

To me the picture has three levels:
1. private/internal, lowest level, host specific libvirt domain xml, can not be accessed from the outside
2. public cluster wide vm domain representation (`VM`) could also be called an instance
3. public cluster wide abstract and persistent vm representation (`VMConfig`)

an instance (`VM`) would always (if it's used) be derived from a `VMConfig´, and this - IMO - makes it easy to understand the transitions, as well as understanding, from which high-level definition a VM definition got derived.